enchant.lic

```diff
@@ -0,0 +1,181 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#enchant
+=end
+
+custom_require.call(%w[common common-arcana events common-crafting events])
+
+class Enchant
+  include DRC
+  include DRCC
+  include DRCA
+
+  def initialize
+    @settings = get_settings
+    @book_type = "artificing"
+    @bag = @settings.crafting_container
+    @bag_items = @settings.crafting_items_in_container
+    @belt = @settings.enchanting_belt
+    arg_definitions = [
+      [
+        { name: 'chapter', regex: /\d+/i, variable: true, description: 'Chapter containing the item.' },
+        { name: 'recipe', display: 'recipe name', regex: /^[A-z\s\-']+$/i, variable: true, description: 'Name of the recipe, wrap in double quotes if this is multiple words.' },
+        { name: 'noun', regex: /\w+/i, variable: true }
+      ]
+    ]
+    #/'  Fix my syntax highlighting
+    
+    Flags.add('enchant-focus', 'material struggles to accept the sigil scribing')
+    Flags.add('enchant-meditate', 'The traced sigil pattern blurs before your eyes')
+    Flags.add('enchant-imbue', 'Once finished you sense an imbue spell will be required to continue enchanting')
+    Flags.add('enchant-push', 'You notice many of the scribed sigils are slowly merging back')
+    Flags.add('enchant-sigil', /You need another ([\w ]*)(primary|secondary) sigil to continue the enchanting process/)
+    Flags.add('enchant-complete', 'With the enchanting process completed, you believe it is safe to collect your things once more.')
+    Flags.add('imbue-failed', 'The streams collide, rending the space before you and disrupting the enchantment')
+    
+    @brazier = "brazier"
+    @use_own_brazier=true
+    DRRoom.room_objs.each do | obj |
+      if obj.include?("enchanter's brazier")
+        @brazier = "enchanter's brazier" 
+        @use_own_brazier = false
+      end
+    end
+    
+    args = parse_args(arg_definitions)
+
+    @item = args.noun
+    @chapter = args.chapter
+    @recipe = args.recipe
+    @primary_sigils = []
+    @secondary_sigils = []
+    
+    @equipment_manager = EquipmentManager.new
+    @equipment_manager.empty_hands
+
+    if @use_own_brazier
+      DRCT.walk_to(@settings.enchanting_room) if @settings.enchanting_room
```

Comment:
💡 You can omit the conditional here (or combine it with the one on line 55) - `walk_to` will safely handle `nil`

Reply:
no need for DRCT either if its included

Reply:
Yeah, we've gone back on forth on whether or not to include the module when we don't have to.

I was originally of the opinion we should omit it whenever possible so make the code simpler, but I've since changed my mind. I think always including the 'prefix' makes it easier for people new to the ecosystem to read the scripts and get a feel for how the shared modules are organized.

Anyway, mine is a weak preference, so I haven't been pushing either way.